test/conformance/ingress/util.go

```diff
@@ -1043,6 +1043,20 @@ func CreateDialContext(ctx context.Context, t *testing.T, ing *v1alpha1.Ingress,
 		t.Fatal("Ingress does not have a public load balancer assigned.")
 	}
 
+	if ing.Status.PublicLoadBalancer.Ingress[0].IP != "" {
+		dial := network.NewBackoffDialer(dialBackoff)
+		ingressIP := ing.Status.PublicLoadBalancer.Ingress[0].IP
+
+		port := 80
+		if ing.Spec.Rules[0].Visibility == v1alpha1.IngressVisibilityExternalIP && ing.Spec.HTTPOption == v1alpha1.HTTPOptionRedirected {
+			port = 443
+		}
+
+		return func(ctx context.Context, _ string, _ string) (net.Conn, error) {
+			return dial(ctx, "tcp", fmt.Sprintf("%s:%d", ingressIP, port))
+		}
+	}
+
```

Comment:
Can we do this as a fallback when DomainInternal isn't present instead of this being the norm

Reply:
eg. with net-contour we always return the IP

Reply:
Updated